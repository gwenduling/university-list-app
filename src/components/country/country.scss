.country-component {
  margin-top: 20px;
}

.result-count {
  font-size: 14px;
}

.show-more {
  width: calc(25% - 10px);

  @media (max-width: 1300px) {
    width: calc(33% - 3px);
  }

  @media (max-width: 500px) {
    width: 100%;
  }
}
