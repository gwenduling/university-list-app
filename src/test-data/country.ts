import { CountryModel } from "../models/countries";
export const mockCountry: CountryModel = {
  alpha2Code: "PH",
  altSpellings: ["PH", "Republic of the Philippines", "Rep√∫blika ng Pilipinas"],
  flag: "https://restcountries.eu/data/phl.svg",
  name: "Philippines",
  nativeName: "Pilipinas",
  count: 0,
  total: 0,
};

export const mockCountry2: CountryModel = {
  alpha2Code: "US",
  altSpellings: ["US", "USA", "United States of America"],
  flag: "https://restcountries.eu/data/phl.svg",
  name: "United States of America",
  nativeName: "United States",
  count: 0,
  total: 0,
};
