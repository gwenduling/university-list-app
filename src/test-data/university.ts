import { UniversityModel } from "../models/universities";

export const mockUniversity: UniversityModel = {
  alpha_two_code: "PH",
  country: "Philippines",
  name: "University Test 1",
  domains: ["university.com"],
  web_pages: ["http://university.com"],
  "state-province": null,
};

export const mockUniversity2: UniversityModel = {
  alpha_two_code: "PH",
  country: "Philippines",
  name: "University Test 2",
  domains: ["university2.com"],
  web_pages: ["http://university2.com"],
  "state-province": null,
};
