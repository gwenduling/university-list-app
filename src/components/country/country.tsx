import { Component } from "react";
import { CountryModel } from "../../models/countries";
import { UniversityModel } from "../../models/universities";

import University from "../university/university";
import CountryItem from "../country-item/country-item";

import "./country.scss";

class Country extends Component<Props> {
  renderUniversity(university: UniversityModel) {
    return (
      <University
        university={university}
        key={university.alpha_two_code + "" + university.name}
      />
    );
  }

  renderButton() {
    return (
      <button
        className="show-more"
        onClick={() => this.props.onClick?.(this.props.country.alpha2Code)}
      >
        show more
      </button>
    );
  }

  render() {
    const universities = this.props.universities
      .filter(
        (university: UniversityModel) =>
          university.alpha_two_code === this.props.country.alpha2Code
      )
      .slice(0, this.props.country.count);

    let button;
    if (this.props.country.count < this.props.country.total) {
      button = this.renderButton();
    } else {
      button = "";
    }

    return (
      <div className="country-component">
        <CountryItem country={this.props.country} />

        <div className="result-count">
          Showing {this.props.country.count} of {this.props.country.total}{" "}
          universities
        </div>

        <div className="universities-list">
          {universities.map((university: UniversityModel) => {
            return this.renderUniversity(university);
          })}
        </div>

        {button}
      </div>
    );
  }
}

interface Props {
  country: CountryModel;
  universities: UniversityModel[];
  onClick?: (code: string) => void;
}
export default Country;
