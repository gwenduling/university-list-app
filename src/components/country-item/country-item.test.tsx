import CountryItem from "./country-item";

import { shallow } from "enzyme";
import { mockCountry } from "../../test-data/country";

describe("Country Item Component", () => {
  it("should match the snapshot", () => {
    const wrapper = shallow(<CountryItem country={mockCountry} />);
    expect(wrapper).toMatchSnapshot();
  });
});
