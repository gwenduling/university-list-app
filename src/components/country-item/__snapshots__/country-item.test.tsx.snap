// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Item Component should match the snapshot 1`] = `
<div
  className="country-item"
>
  <img
    alt="Philippines flag"
    className="flag"
    src="https://restcountries.eu/data/phl.svg"
  />
  <div
    className="country-name"
  >
    <a
      className="name"
      href="/countries/PH"
    >
      Philippines
    </a>
    <p
      className="native"
    >
      (
      Pilipinas
      )
    </p>
  </div>
</div>
`;
